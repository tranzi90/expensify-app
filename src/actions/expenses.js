// import uuid from "uuid"
import database from "../firebase"

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
})

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData

        const expense = { description, note, amount, createdAt}

        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }))
        })
    }
}

export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',
    id
})

export const startRemoveExpense = (id) => {
    return async (dispatch) => {
        await database.ref(`expenses/${id}`).remove()

        dispatch(removeExpense(id))
    }
}

export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})

export const startEditExpense = (id, updates) => {
    return async (dispatch) => {
        await database.ref(`expenses/${id}`).update(updates)

        dispatch(editExpense(id, updates))
    }
}

export const setExpenses = (expenses) => ({
    type: 'SET_EXPENSES',
    expenses
})

export const startSetExpenses = () => {
    return async (dispatch) => {
        const snapshot = await database.ref(`expenses`).once('value')

        let expenses = []
        snapshot.forEach((child) => {
            expenses.push({
                id: child.key,
                ...child.val()
            })
        })

        dispatch(setExpenses(expenses))
    }
}