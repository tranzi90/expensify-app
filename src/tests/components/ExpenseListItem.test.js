import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import React from 'react'
import expenses from '../fixtures/expenses'

it('should render ExpenseListItem with fixture data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)

    expect(wrapper).toMatchSnapshot()
})
