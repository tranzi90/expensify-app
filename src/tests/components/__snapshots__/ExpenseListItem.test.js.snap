// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with fixture data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      жовка
    </h3>
  </Link>
  <p>
    195
     - 
    0
  </p>
</div>
`;
